Objective-C/CBLReplicator.mm

```diff
@@ -502,15 +498,26 @@ - (void) c4StatusChanged: (C4ReplicatorStatus)c4Status {
         #if TARGET_OS_IPHONE
             [self endBackgrounding];
         #endif
-            if (_conflictCount > 0) {
-                CBLLogInfo(Sync, @"%@: Stopped but waiting for conflict resolution (pending = %d) "
-                                  "to finish before notifying.", self, _conflictCount);
-                _deferStoppedNotification = YES;
-                _state = kCBLStateStopping; // Will be stopped after all conflicts are resolved
-            } else
+            
+            if (_conflictCount == 0)
                 [self stopped];
+        }
+        
+        // replicator status callback
+        /// stopped and idle state, we will defer status callback till conflicts finish resolving
+        if (_conflictCount > 0 && (c4Status.level == kC4Stopped || c4Status.level == kC4Idle)) {
+            CBLLogInfo(Sync, @"%@: Status = %d, but waiting for conflict resolution (pending = %d) "
+            "to finish before notifying.", self, c4Status.level, _conflictCount);
+            
+            _deferReplicatorNotification = YES;
+            if (c4Status.level == kC4Stopped)
+                _state = kCBLStateStopping;
         } else
+            _deferReplicatorNotification = NO;
```

Comment:
##### Copied comments from the closed PR
> It's also possible that the IDLE status could be changed to Busy so the deferred IDLE might not valid anymore. For that case, we can reset _deferReplicatorNotification = NO.

> Please also review other scenarios of the status change from IDLE to something else to see the behavior of deferring the idle notification.

When there is no conflicts or any other state change will reset the `_deferReplicatorNotification`